pydantic/_internal/_schema_generation_shared.py

```diff
@@ -12,64 +12,12 @@
 if TYPE_CHECKING:
     from ..json_schema import GenerateJsonSchema, JsonSchemaValue
     from ._core_utils import CoreSchemaOrField
+    from ._generate_schema import GenerateSchema
 
     GetJsonSchemaFunction = Callable[[CoreSchemaOrField, GetJsonSchemaHandler], JsonSchemaValue]
     HandlerOverride = Callable[[CoreSchemaOrField], JsonSchemaValue]
 
 
-class UnpackedRefJsonSchemaHandler(GetJsonSchemaHandler):
-    """
-    A GetJsonSchemaHandler implementation that automatically unpacks `$ref`
-    schemas so that the caller doesn't have to worry about that.
-
-    This is used for custom types and models that implement `__get_pydantic_core_schema__`
-    so they always get a `non-$ref` schema.
-
-    Used internally by Pydantic, please do not rely on this implementation.
-    See `GetJsonSchemaHandler` for the handler API.
-    """
```

Comment:
This functionality was half broken before and is fully broken now. We're just going to have to face the fact that if you are messing with schemas you'll have to deal with refs.